src/care/algorithm_impl.h

```diff
@@ -471,7 +471,7 @@ CARE_HOST_DEVICE CARE_INLINE int BinarySearch(const T *map, const int start,
 
    while (khi-klo > 1) {
       k = (khi+klo) >> 1 ;
-      if (map[k] == num) {
+      if (! (map[k] < num) && !(num < map[k])) {
```

Comment:
can you add a comment somewhere in this routine indicating that it should only use <?